### PR TITLE
feat: ML challenge page routing

### PR DESCRIPTION
**Pull request by codemonkey800**
#705

- Adds routing for the ML challenge page at `/competition` (pending confirmation from @junxini)
- Sets up skeleton page for ML challenge page with footer and header layout
- Adds feature flag + redirect to 404 page when flag is disabled

## Demo

image

### CODE REVIEW

File: frontend/packages/data-portal/app/components/Layout/TopNavigation.tsx [hunk 1]
```diff
@@ -1,15 +1,40 @@
...
+interface TopNavLink {
+  isActive(pathname: string): boolean
+  label: I18nKeys
+  link: string
+}
+
+const TOP_NAV_LINKS: TopNavLink[] = [
+  {
+    isActive: (pathname) =>
+      pathname.includes('/datasets') || pathname.includes('/runs'),
```
Comment (tihuan) [inline frontend/packages/data-portal/app/components/Layout/TopNavigation.tsx:22]: General, non blocking, out of scope question: 

In Cell Science, we use a `routes.ts` file to hold route constants, so we have single source of truth, like so:

```ts
export enum ROUTES {
  HOMEPAGE = "/",
  CENSUS_DIRECTORY = "/census-models",
  COLLECTION = "/collections/:id",
  COLLECTIONS = "/collections",
  ...
}
```

Is it something worth doing here as a separate PR 👌 ?

**Review by tihuan — APPROVED** [review]
LGTM! Really cool to see `i18n` being used in an app! I've never done it before 😆 👏

File: frontend/packages/data-portal/app/components/Layout/TopNavigation.tsx [hunk 2]
```diff
@@ -24,18 +49,22 @@ export function TopNavigation() {
...
-        to="/browse-data/datasets"
-      >
-        {i18n.browseData}
-      </Link>
+      {TOP_NAV_LINKS.filter(
+        (link) => showMlChallenge || link.label !== 'competition',
+      ).map((link) => (
+        <Link
+          className={cns(
+            'text-sds-header-s font-semibold mr-sds-xxl p-0',
```
Comment (kne42) [inline frontend/packages/data-portal/app/components/Layout/TopNavigation.tsx:57]:
```suggestion
            'text-sds-header-s leading-sds-header-s font-semibold mr-sds-xxl p-0',
```

**Review by kne42 — APPROVED** [review]
LGTM

**Pull request merged**